src/app/hooks/env.ts

```diff
@@ -17,16 +19,35 @@ export const useActiveProfile = (): Contracts.IProfile => {
 		}
 
 		return context.env.profiles().findById(profileId);
-	}, [context, profileId]); // eslint-disable-line react-hooks/exhaustive-deps
+	}, [context.env, history.location.pathname, profileId]);
 };
 
-export const useActiveWallet = () => {
+export const useActiveWallet = (): Contracts.IReadWriteWallet => {
 	const profile = useActiveProfile();
 	const { walletId } = useParams<{ walletId: string }>();
 
 	return useMemo(() => profile.wallets().findById(walletId), [profile, walletId]);
 };
 
+export const useNetworks = (profile?: Contracts.IProfile) => {
+	const activeProfile = useActiveProfile();
```

Comment:
```suggestion
export const useNetworks = () => {
	const activeProfile = useActiveProfile();
```
I think the `profile` isn't needed as a parameter here, because other profiles also come from `useActiveProfile`, which has already been used here. (e.g. [here](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/dashboard/components/Wallets/Wallets.tsx#L37-L42) & [here](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/vote/hooks/use-vote-filters.ts#L20))

Reply:
I integrated this hook with `useWalletFilters` which has its own profile, that could change in `useVoteFilters` for example. Also, unfortunately, `useWalletFilters` hook tests do not provide a profile from `usePath`

Reply:
there are 2 places that are using `useWalletFilters`
in `useVoteFilters` and `Wallets.tsx` that both are passing profile from `useActiveProfile`

[Wallets](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/dashboard/components/Wallets/Wallets.tsx#L37-L42)
[useVoteFilters](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/vote/hooks/use-vote-filters.ts#L20)  => [Votes](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/vote/pages/Votes/Votes.tsx#L55)

we need the profile in `useWalletFilters`, but we don't need it [here](https://github.com/PayvoHQ/wallet/blob/b6ae7d240e7f31d656dcbe2ed2ec1105d8a750c1/src/domains/dashboard/components/FilterWallets/hooks.ts#L13) and [here](https://github.com/PayvoHQ/wallet/blob/b6ae7d240e7f31d656dcbe2ed2ec1105d8a750c1/src/app/hooks/env.ts#L32), because it's already getting profile [from here](https://github.com/PayvoHQ/wallet/blob/b6ae7d240e7f31d656dcbe2ed2ec1105d8a750c1/src/app/hooks/env.ts#L33)

Reply:
Ok, got it